test/unit-test/twcc_manager/twcc_manager_utest.c

```diff
@@ -176,22 +462,223 @@ void test_twccFindPacket_Found( void )
+
+    result = RtcpTwccManager_HandleTwccPacket( &( twccManager ),
+                                               &( twccPacket ),
+                                               &( twccBandwidthInfo ) );
+    TEST_ASSERT_EQUAL( RTCP_TWCC_MANAGER_RESULT_OK,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate Twcc Manager Handle packet functionality.
+ */
+void test_twccHandlePacket( void )
+{
+    RtcpTwccManager_t twccManager;
+    RtcpTwccManagerResult_t result;
+    RtcpTwccPacket_t twccPacket;
+    TwccBandwidthInfo_t bandwidthInfo = { 0 };
+    TwccPacketInfo_t packetInfo = { 0 };
+    PacketArrivalInfo_t arrivalInfoList[TWCC_PACKET_INFO_ARRAY_LENGTH];
+    uint64_t localSentTime = time( NULL );
+    uint16_t seqNum = 256;
+    size_t i;
+
+    result = RtcpTwccManager_Init( &( twccManager ),
+                                   &( twccPacketInfoArray[ 0 ] ),
+                                   TWCC_PACKET_INFO_ARRAY_LENGTH );
+    TEST_ASSERT_EQUAL( RTCP_TWCC_MANAGER_RESULT_OK,
+                       result );
+
+    for(i = 0; i < TWCC_PACKET_INFO_ARRAY_LENGTH; i++)
+    {
+        packetInfo.packetSize = i;
+        packetInfo.localSentTime = localSentTime + ( int64_t )i;
+        packetInfo.packetSeqNum = seqNum - 1 + ( uint16_t )( i );
+        result = RtcpTwccManager_AddPacketInfo( &( twccManager ),
+                                                &( packetInfo ) );
+        TEST_ASSERT_EQUAL( RTCP_TWCC_MANAGER_RESULT_OK,
+                           result );
+
+        arrivalInfoList[i].seqNum = packetInfo.packetSeqNum;
+        if( i % 2 == 0 ) {
+            arrivalInfoList[i].remoteArrivalTime = RTCP_TWCC_PACKET_LOST_TIME; // Set some packets as lost
+        }
+        else {
+            arrivalInfoList[i].remoteArrivalTime = localSentTime + ( int64_t )i; // Set some packets as received
+        }
+    }
+
+    // Creating a valid RtcpTwccPacket_t
+    twccPacket.arrivalInfoListLength = TWCC_PACKET_INFO_ARRAY_LENGTH;
+    twccPacket.pArrivalInfoList = arrivalInfoList;
+
+    result = RtcpTwccManager_HandleTwccPacket( &( twccManager ),
+                                               &( twccPacket ),
+                                               &bandwidthInfo );
+
+    TEST_ASSERT_EQUAL( RTCP_TWCC_MANAGER_RESULT_OK,
+                       result );
+    TEST_ASSERT_GREATER_THAN( 0,
+                              bandwidthInfo.duration );
+    TEST_ASSERT_EQUAL( TWCC_PACKET_INFO_ARRAY_LENGTH,
+                       bandwidthInfo.sentPackets );
+    TEST_ASSERT_GREATER_THAN( 0,
+                              bandwidthInfo.sentBytes );
+    TEST_ASSERT_GREATER_THAN( 0,
+                              bandwidthInfo.receivedBytes );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate Twcc Manager Handle packet functionality when unidentified Sequence Number Packet is found.
```

Comment:
Similar issue here. What is unidentified seq num packet? And why does RtcpTwccManager_HandleTwccPacket() return true in this case?

Reply:
TwccPacket is a control Packet information that the remote gets. Thus there may be circumstances that it needs a seq -1 number packet. In such case, this unit-test cover that circumstance.